lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure.hs

```diff
@@ -345,23 +366,103 @@ wonders interval =
 
 {-----------------------------------------------------------------------------
     Operations
-    Writing to blockchain
+    Constructing transactions
 ------------------------------------------------------------------------------}
+data ErrCreatePayment
+    = ErrCreatePaymentNotRecentEra (Read.EraValue Read.Era)
+    | ErrCreatePaymentBalanceTx (Write.ErrBalanceTx Write.Conway)
+    deriving (Eq, Show)
 
+-- | Create a payment to a list of destinations.
 createPayment
-    :: [(Address, Write.Value)] -> WalletState -> Maybe Write.TxBody
-createPayment = undefined
+    :: Read.EraValue Read.PParams
+    -> Write.TimeTranslation
+    -> [(Address, Write.Value)]
+    -> WalletState
+    -> Either ErrCreatePayment Write.Tx
+createPayment (Read.EraValue (Read.PParams pparams :: Read.PParams era)) a b w =
+    case Read.theEra :: Read.Era era of
+        Read.Conway ->
+            first ErrCreatePaymentBalanceTx
+                $ createPaymentConway pparams a b w
+        era' -> Left $ ErrCreatePaymentNotRecentEra (Read.EraValue era')
+
+-- | In the Conway era: Create a payment to a list of destinations.
+createPaymentConway
+    :: Write.PParams Write.Conway
+    -> Write.TimeTranslation
+    -> [(Address, Write.Value)]
+    -> WalletState
+    -> Either (Write.ErrBalanceTx Write.Conway) Write.Tx
+createPaymentConway pparams timeTranslation destinations w =
```

Comment:
I generally agree on the `IsRecentEra era` polymorphism.

However, for the Deposit Wallet MVP, I figured that I would limit the scope to `Conway` — that's why `Cardano.Wallet.Deposit.Write.Tx` is pinned to a single era for now.

As for the actual implementation of era-polymorphic functions: It's a matter of code reuse — the function `createPaymentConway` can be reused in an era-polymorphic implementation, or it can be cheaper to write `createPaymentPolymorphic` directly.

Compare

```hs
fooPolymorphic :: IsRecentEra era => Thing era
fooPolymorphic = … barConway … barDonway …

barConway :: Otherthing Conway
barDonway :: Otherthing Donway
```

vs

```hs
fooPolymorphic :: IsRecentEra era => Thing era
fooPolymorphic = … barPolymorphic …

barPolymorphic :: IsRecentEra era => Otherthing Conway era
```

The second style tends to use lines of code more effectively, but not always due to case distinctions and type inference. The first style can always be wrapped into the second style by adding code rather than changing it.